topics/go/algorithms/data/bst/bst.go

```diff
@@ -0,0 +1,113 @@
+package bst
+
+import "fmt"
+
+// TODO This is currently a draft...
+
+// Node represents the data being stored.
+type Node struct {
+	Data  int
+	left  *Node
+	right *Node
+}
+
+// BST represents a binary search tree.
+type BST struct {
+	root *Node
+}
+
+// New creates a new BST.
+func New() *BST {
+	return &BST{
+		root: nil,
+	}
+}
+
+// Insert an integer into the BST.
+func (bst *BST) Insert(value int) {
+
+	// Create a new node.
+	n := &Node{
+		Data:  value,
+		left:  nil,
+		right: nil,
+	}
+
+	// If the root node is nil set the root node
+	// to n
+	if bst.root == nil {
+		bst.root = n
+	} else {
+		insertNode(bst.root, n)
+	}
+}
+
+// Max returns the max int
+func (bst *BST) Max() (int, error) {
+
+	currentNode := bst.root
+
+	if currentNode == nil {
+		return 0, fmt.Errorf("root node is nil")
+	}
+
+	for {
+		if currentNode.right == nil {
```

Comment:
This loops until the right side is nil. Wondering if that can be incorporated into the for.  I could be wrong.